src/views/User.vue

```diff
@@ -204,4 +260,49 @@ const summarize = courses => {
     return message
   }
 }
+
+const onFutureCoursesPreferenceChange = (value) => {
+  const optInNewCourses = value === 'all'
+  updateOptInNewCourses(uid, {optInNewCourses})
+    .then((prefs) => {
+      const {optInNewCourses: optAll, doNotEmail} = prefs || {}
+      futureCoursesPref.value = optAll ? 'all' : 'choose'
+      user.value.optInNewCourses = !!optAll
+
+      if (typeof doNotEmail === 'boolean') {
+        user.value.doNotEmail = doNotEmail
+        emailReceive.value = doNotEmail
+      }
+      alertScreenReader('Future courses preference updated.')
+    })
+    .catch(() => {
+      // revert to server-known state
+      futureCoursesPref.value = user.value.optInNewCourses ? 'all' : 'choose'
+      alertScreenReader('Failed to update future courses preference.')
+    })
+}
+
+const onEmailReceiveChange = (value, {force = false} = {}) => {
+  if (isEmailDisabled.value && !force) {
+    emailReceive.value = false
+    return
+  }
+  updateDoNotEmail(uid, {doNotEmail: !value})
+    .then((prefs) => {
+      if (typeof prefs?.doNotEmail === 'boolean') {
+        user.value.doNotEmail = !prefs.doNotEmail
+        emailReceive.value = !prefs.doNotEmail
+      }
+      if (typeof prefs?.optInNewCourses === 'boolean') {
+        user.value.optInNewCourses = prefs.optInNewCourses
+        futureCoursesPref.value = prefs.optInNewCourses ? 'all' : 'choose'
+      }
+      alertScreenReader('Email preference updated.')
+    })
+    .catch(() => {
+      emailReceive.value = !user.value.doNotEmail
+      alertScreenReader('Failed to update email preference.')
```

Comment:
@mohamalnadi - Does the catch block above silence fatal errors? Would the user be aware that the update failed?